addons/account/wizard/account_payment_register.py

```diff
@@ -191,7 +191,11 @@ def _get_line_batch_key(self, line):
 
         partner_bank_account = self.env['res.partner.bank']
         if move.is_invoice(include_receipts=True):
-            partner_bank_account = move.partner_bank_id._origin
+            if move.move_type == 'in_refund':
+                bank_ids = move.commercial_partner_id.bank_ids.filtered(lambda bank: bank.company_id in (False, move.company_id))
+                partner_bank_account = bank_ids and bank_ids[0]
+            else:
+                partner_bank_account = move.partner_bank_id._origin
```

Comment:
Damn, I wrote a message but didn't send it for this one:

Could you clarify why this part is required?
From my understanding, you are already set the right `move.partner_bank_id` in the compute method you modified and thus, it shouldn't need this part.

Reply:
With the modified calculation method, we have two possibilities for `move.partner_bank_id` : 
- for an inbound : `move.partner_bank_id` will correspond to the one of the company
- for an outbound : `move.partner_bank_id` will correspond to the one of the customer.

About this specific part of fix :
When making a batch payment with these compute changes, we can't directly use `move.partner_bank_id` since it matches the customer's bank for an outbound and company's one for an inbound. So it won't group the payments like you pointed out here (https://github.com/odoo/odoo/pull/85982#issuecomment-1073821693) for the first fix I proposed.
To prevent that and keep batch payments, I'm using company's bank in case of an outbound so that it can be grouped with corresponding inbound.